frontstage/controllers/collection_exercise_controller.py

```diff
@@ -40,26 +46,56 @@ def get_collection_exercise_event(collection_exercise_id, tag):
         raise ApiError(logger, response,
                        collection_exercise_id=collection_exercise_id,
                        log_level='warning' if response.status_code == 404 else 'exception',
-                       message='Failed to retrieve collection exercise event',
-                       tag=tag)
+                       message='Failed to retrieve collection exercise events')
 
-    logger.debug('Successfully retrieved collection exercise event', collection_exercise_id=collection_exercise_id, tag=tag)
+    logger.debug('Successfully retrieved collection exercise events', collection_exercise_id=collection_exercise_id)
     return response.json()
 
 
-def get_collection_exercise_events(collection_exercise_id):
-    logger.debug('Attempting to retrieve collection exercise events', collection_exercise_id=collection_exercise_id)
-    url = f"{app.config['COLLECTION_EXERCISE_URL']}/collectionexercises/{collection_exercise_id}/events"
+def get_collection_exercises_for_survey(survey_id):
+    logger.debug('Retrieving collection exercises for survey', survey_id=survey_id)
+
+    url = f"{app.config['COLLECTION_EXERCISE_URL']}/collectionexercises/survey/{survey_id}"
 
     response = requests.get(url, auth=app.config['COLLECTION_EXERCISE_AUTH'])
 
     try:
         response.raise_for_status()
     except requests.exceptions.HTTPError:
         raise ApiError(logger, response,
-                       collection_exercise_id=collection_exercise_id,
+                       survey_id=survey_id,
                        log_level='warning' if response.status_code == 404 else 'exception',
-                       message='Failed to retrieve collection exercise events')
-
-    logger.debug('Successfully retrieved collection exercise events', collection_exercise_id=collection_exercise_id)
-    return response.json()
+                       message='Failed to retrieve collection exercises for survey')
+
+    logger.debug("Successfully retrieved collection exercises for survey", survey_id=survey_id)
+    collection_exercises = response.json()
+
+    for collection_exercise in collection_exercises:
+        if collection_exercise['events']:
+            collection_exercise['events'] = convert_events_to_new_format(collection_exercise['events'])
+
+    return collection_exercises
+
+
+def enrolment_with_collection_exercise(enrolment, collection_exercises):
```

Comment:
Might want to name this function `link_enrolment_with_collection_exercise` to distinguish it better from `get_enrolments_with_collection_exercises`